app/uk/gov/hmrc/apiplatformoutboundsoap/repositories/OutboundMessageRepository.scala

```diff
@@ -118,8 +118,8 @@ class OutboundMessageRepository @Inject()(mongoComponent: MongoComponent, appCon
       .toFutureOption()
   }
 
-  def findById(messageId: String): Future[Option[OutboundSoapMessage]] = {
-    collection.find(filter = equal("messageId", messageId)).headOption()
+  def findById(searchForId: String): Future[Option[OutboundSoapMessage]] = {
```

Comment:
This method is already being used by ConfirmationService to process the request wherein the id is strictly messageId. Now that it could be globalId, would it break anything ?

Reply:
That would require the database to contain a record with a globalId (which is random UUID) which matched another record's messageId (which we leave the ICS2 team to generate themselves). I figured this is a vanishingly small risk so opted to keep the repository code simpler.